File: app/src/main/java/calculator/simplemobiletools/com/simple_calculator/Formatter.java
package calculator.simplemobiletools.com.simple_calculator;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.util.Locale;

public class Formatter {
    public static String doubleToString(double d) {
        if (d == (long) d) {
            return String.format("%d", (long) d);
        } else {
            final DecimalFormatSymbols symbols = new DecimalFormatSymbols(Locale.US);
            symbols.setDecimalSeparator('.');

            final DecimalFormat formatter = new DecimalFormat();
            formatter.setMaximumIntegerDigits(12);
            formatter.setMaximumFractionDigits(12);
            formatter.setDecimalFormatSymbols(symbols);
            formatter.setGroupingUsed(false);
            return formatter.format(d);
        }
    }
}

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/Formatter.kt
package com.simplemobiletools.calculator.helpers

import java.math.BigDecimal
import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.*

object Formatter {
    fun bigDecimalToString(d: BigDecimal): String {
        val symbols = DecimalFormatSymbols(Locale.US)
        symbols.decimalSeparator = '.'
        symbols.groupingSeparator = ','

        val formatter = DecimalFormat()
        formatter.maximumFractionDigits = 50
        formatter.decimalFormatSymbols = symbols
        formatter.isGroupingUsed = true
        return formatter.format(d)
    }

    fun stringToBigDecimal(str: String): BigDecimal = BigDecimal(str.replace(",", ""))
}

fun BigDecimal.format(): String = Formatter.bigDecimalToString(this)
